sklearn/cluster/dbscan_.py

```diff
@@ -111,7 +114,14 @@ def dbscan(X, eps=0.5, min_samples=5, metric='minkowski',
         neighbors_model.fit(X)
         neighborhoods = neighbors_model.radius_neighbors(X, eps,
                                                          return_distance=False)
-        neighborhoods = np.array(neighborhoods)
+
+    # Ugly hack to make dtype=object, ndim=1 array when all neighborhoods
```

Comment:
Hopefully this will be fixed in neighbors module soon. But what's wrong with
```
tmp = np.empty(neighborhoods.shape[0], dtype='O')
tmp[:] = neighborhoods.tolist()
neighborhoods = tmp
```
it's a little bit clearer, perhaps?